hardware/arduino/avr/cores/arduino/USBCore.cpp

```diff
@@ -72,18 +65,14 @@ const u8 STRING_PRODUCT[] PROGMEM = USB_PRODUCT;
 const u8 STRING_MANUFACTURER[] PROGMEM = USB_MANUFACTURER;
 
 
-#ifdef CDC_ENABLED
 #define DEVICE_CLASS 0x02
-#else
-#define DEVICE_CLASS 0x00
-#endif
 
 //	DEVICE DESCRIPTOR
 const DeviceDescriptor USB_DeviceDescriptor =
 	D_DEVICE(0x00,0x00,0x00,64,USB_VID,USB_PID,0x100,IMANUFACTURER,IPRODUCT,0,1);
 
-const DeviceDescriptor USB_DeviceDescriptorA =
-	D_DEVICE(DEVICE_CLASS,0x00,0x00,64,USB_VID,USB_PID,0x100,IMANUFACTURER,IPRODUCT,0,1);
+const DeviceDescriptor USB_DeviceDescriptorB =
+	D_DEVICE(0xEF,0x02,0x01,64,USB_VID,USB_PID,0x100,IMANUFACTURER,IPRODUCT,0,1);
```

Comment:
I might get this wrong, but you have to add 0xEF, 0x02 and 0x01 to the upper Descriptor. Thats how I did it in my source at least.

Also I'd go for definitions of those values so they are better understandable.